node-runner-cli/config/SystemDConfig.py

```diff
@@ -0,0 +1,157 @@
+    gateway_settings: None
+
+    def __iter__(self):
+        class_variables = {key: value
+                           for key, value in self.__class__.__dict__.items()
+                           if not key.startswith('__') and not callable(value)}
+        for attr, value in class_variables.items():
+            if attr in ['keydetails']:
+                yield attr, dict(self.__getattribute__(attr))
+            elif self.__getattribute__(attr):
+                yield attr, self.__getattribute__(attr)
+
+    def to_yaml(self):
+        config_to_dump = dict(self)
+        config_to_dump["core_node_settings"] = dict(self.core_node_settings)
+        config_to_dump["core_node_settings"]["keydetails"] = dict(self.core_node_settings.keydetails)
+        config_to_dump["common_settings"] = dict(self.common_settings)
+        config_to_dump["common_settings"]["nginx_settings"] = dict(self.common_settings.nginx_settings)
+        return yaml.dump(config_to_dump, sort_keys=False, default_flow_style=False, explicit_start=True,
+                         allow_unicode=True)
+
+    def to_file(self, config_file):
+        config_to_dump = dict(self)
+        config_to_dump["core_node_settings"] = dict(self.core_node_settings)
+        config_to_dump["core_node_settings"]["keydetails"] = dict(self.core_node_settings.keydetails)
+        config_to_dump["common_settings"] = dict(self.common_settings)
+        config_to_dump["common_settings"]["nginx_settings"] = dict(self.common_settings.nginx_settings)
+        with open(config_file, 'w') as f:
+            yaml.dump(config_to_dump, f, sort_keys=True, default_flow_style=False)
+
+
+def from_dict(dictionary: dict) -> SystemDSettings:
+    settings = SystemDSettings({})
+    settings.core_node_settings = CoreSystemdSettings(dictionary["core_node_settings"])
+    settings.core_node_settings.keydetails = KeyDetails(dictionary["core_node_settings"]["keydetails"])
+    settings.common_settings = CommonSystemdSettings(dictionary["common_settings"])
+    settings.common_settings.nginx_settings = SystemdNginxConfig(dictionary["common_settings"]["nginx_settings"])
+    return settings
+
+
+def extract_network_id_from_arg(network_id_arg) -> int:
+    if network_id_arg == 1 or network_id_arg == 2:
+        return network_id_arg
+    elif network_id_arg in ["2", "s", "S", "stokenet"]:
+        return 2
+    elif network_id_arg in ["1", "m", "M", "mainnet"]:
+        return 1
+    elif network_id_arg is None or network_id_arg == "":
+        return None
+    else:
+        print(
+            "Not a valid argument for network id. Please enter either '1' 'm' 'M' 'mainnet' or '2' 's' 'S' 'stokenet'")
+        sys.exit(1)
+
+
+def validate_network_id(network_prompt):
+    if network_prompt.lower() in ["s", "S", "stokenet"]:
+        network_id = 2
+    elif network_prompt.lower() in ["m", "M", "mainnet"]:
+        network_id = 1
+    elif network_prompt in ["1", "2", "3", "4", "5", "6", "7", "8"]:
```

Comment:
The network id's are changed for babylon testnets. 
https://github.com/radixdlt/babylon-node/blob/4e6ef24f3dff750ea3515416e27bcaa8f454cbdc/common/src/main/java/com/radixdlt/networks/Network.java#L86